gemini/sample-apps/genwealth/function-scripts/write-metadata/main.py

```diff
@@ -0,0 +1,63 @@
+"""Function to write metadata for new pdf documents"""
+
+import os
+from pathlib import Path
+import uuid
+
+import functions_framework
+from google.cloud import storage
+
+
+@functions_framework.cloud_event
+def write_metadata(cloud_event):
+    """Main function"""
+    # Set event vars
+    data = cloud_event.data
+    event_id = cloud_event["id"]
+    event_type = cloud_event["type"]
+    bucket = data["bucket"]
+    name = data["name"]
+    metageneration = data["metageneration"]
+    timeCreated = data["timeCreated"]
+    updated = data["updated"]
+
+    # Print event vars to log
+    print(f"Event ID: {event_id}")
+    print(f"Event type: {event_type}")
+    print(f"Bucket: {bucket}")
+    print(f"File: {name}")
+    print(f"Metageneration: {metageneration}")
+    print(f"Created: {timeCreated}")
+    print(f"Updated: {updated}")
+
+    # Set local vars
+    project_id = os.environ["PROJECT_ID"]
+    metadata_bucket = "{}-docs-metadata".format(project_id)
+    uid = str(uuid.uuid4())
+    ticker = Path(name).stem
+    target_file_name = "{}.jsonl".format(ticker)
+
+    # Build metadata jsonl
+    metadata = (
```

Comment:
Can this be done in a dictionary, then converted to jsonl? This is difficult to read.

Reply:
Will revisit this.

Reply:
Moving to backlog.